rs/nns/integration_tests/src/create_service_nervous_system.rs

```diff
@@ -181,12 +181,115 @@ fn test_several_proposals() {
     assert_eq!(snses.len(), 1, "{:#?}", snses);
 }
 
+#[test]
+fn test_nf_is_not_permitted() {
+    // Step 1: Prepare the world.
+
+    let state_machine = state_machine_builder_for_nns_tests().build();
+
+    // Step 1.1: Boot up NNS.
+    let nns_init_payload = NnsInitPayloadsBuilder::new()
+        .with_initial_invariant_compliant_mutations()
+        .with_test_neurons_fund_neurons(100_000_000_000_000)
+        .with_sns_dedicated_subnets(state_machine.get_subnet_ids())
+        .with_sns_wasm_access_controls(true)
+        // TODO: Delete this once the SNS_WASM canister takes any requests
+        // coming from NNS governance.
+        .with_sns_wasm_allowed_principals(vec![PrincipalId::from(GOVERNANCE_CANISTER_ID)])
+        .build();
+    // Note that this uses governance with cfg(features = "test") enabled.
+    setup_nns_canisters_with_features(&state_machine, nns_init_payload, /* features */ &[]);
+    add_real_wasms_to_sns_wasms(&state_machine);
+    let dapp_canister = create_canister_id_at_position(&state_machine, 1000, None);
+    set_controllers(
+        &state_machine,
+        PrincipalId::new_anonymous(),
+        dapp_canister,
+        vec![ROOT_CANISTER_ID.get()],
+    );
+
+    // In real life, DFINITY would top up SNS_WASM's cycle balance (and the SNS
+    // is supposed to repay with ICP raised).
+    state_machine.add_cycles(SNS_WASM_CANISTER_ID, 200 * ONE_TRILLION);
+
+    // Step 2: Run code under test. Inspect intermediate results.
+
+    // Step 2.1: Make a proposal.  Should fail because it's using NF funding on non-test gov build.
+    let response_1 = make_proposal(&state_machine, /* sns_number = */ 1, true);
+    if let Some(manage_neuron_response::Command::MakeProposal(_)) = response_1.command {
+        panic!("Should not be able to submit a propsals with NF matched funding enabled.");
+    }
```

Comment:
NCR

This could be a little stronger by asserting that response is X, rather than asserting that the response is not Y.

Reply:
done.